### ISSUE TITLE
Add appendix about the use of EAD

### ISSUE THREAD

**Issue by gselander** [lake-wg/edhoc#210]

**Comment by emanjon** [comment]
Agree. This is releated to #149. I should not have closed that one. EAD is stil underspecified.

**Comment by gselander** [comment]
Related to Stephen's comment in #202:

> 5.3.3: is it ok to pass EAD_2 to the application before
   checking authentication?

If EAD_2 indeed contains external authorization data (e.g. information about identity of intended endpoint), then EAD_2 needs to be processed before you can verify the identity of the other endpoint and the integrity of the message. But if we look at EAD_3, the same things apply, but we also claim that EAD_3 is protected between Initiator and Responder, which with the current order of things isn't verified at the time when EAD processed.

**Comment by gselander** [comment]
PR #240 addresses this issue. 

* Rewrite of section 3.8 and added appendix with more details
* Application should consider EAD unprotected by EDHOC.
* Details of how EAD protection is moved to security considerations (8.1)

**Comment by gselander** [comment]
#240 as above and with the new appendix E is now merged.
